ha/core/controllers/pcs/node_controller.py

```diff
@@ -236,3 +371,73 @@ def shutdown(self, nodeid: str) -> dict:
                 status: Succeeded, Failed, InProgress
         """
         raise HAUnimplemented("This operation is not implemented.")
+
+    @controller_error_handler
+    def stop(self, node_id: str, timeout: int = -1, **op_kwargs) -> dict:
+        """
+        Stop Cluster on node with node_id.
+        Args:
+            node_id (str): Private fqdn define in conf store.
+        Returns:
+            ([dict]): Return dictionary. {"status": "", "msg":""}
+                status: Succeeded, Failed, InProgress
+        """
+        check_cluster = op_kwargs.get("check_cluster") if op_kwargs.get("check_cluster") is not None else True
+        poweroff = op_kwargs.get("poweroff") if op_kwargs.get("poweroff") is not None else False
+        storageoff = op_kwargs.get("storageoff") if op_kwargs.get("storageoff") is not None else False
+        try:
+            # Raise exception if user does not have proper permissions
+            self._validate_permissions()
+            # Raise exception if node_id is not valid
+            self.validate_node(node_id=node_id)
+
+            # Get the nodeid from pvtfqdn
+            nodeid = self._health_manager.get_key(f"{const.PVTFQDN_TO_NODEID_KEY}/{node_id}")
+
+            # stop all services running on node with node_id
+            node_status = self._system_health.get_node_status(node_id=nodeid).get("status")
+            if node_status == HEALTH_STATUSES.OFFLINE.value:
+                Log.info(f"For stop {node_id}, Node already in offline state.")
+                status = f"Node {node_id} is already in offline state."
+                return {"status": const.STATUSES.SUCCEEDED.value, "output": node_status, "error": "", "msg": status}
+            elif node_status == HEALTH_STATUSES.FAILED.value:
+                raise ClusterManagerError(f"Failed to stop {node_id}."
+                                          f"node is in {node_status} state.")
+            else:
+                if check_cluster:
+                    # Checks whether cluster is going to be offline if node with node_id is stopped.
+                    res = self.check_cluster_feasibility(nodeid=node_id)
+                    if res.get("status") == const.STATUSES.FAILED.value:
+                        return res
```

Comment:
return in same format as mentioned on line 402

Reply:
These res itself is in same format "return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Stopping the node will cause a loss of the quorum"}". So i think no need to format it again.